api/service.go

```diff
@@ -499,20 +525,21 @@ func (s *service) processFindRequest(req *pb.FindRequest, findFunc func(q *db.JS
 	return &pb.FindReply{Instances: byteInstances}, nil
 }
 
-func (s *service) getDB(idStr string) (*db.DB, error) {
-	id, err := uuid.Parse(idStr)
+func (s *Service) getDB(idStr string) (d *db.DB, id thread.ID, err error) {
```

Comment:
Nice, been thinking I need to start using named return values so the code is more self documenting.

Reply:
Although, if self documenting code is what we're going for, I guess names better than `d` and `id` would be in order.

Reply:
I've been liking the named returns when one of them is not a pointer receiver, i.e., `string` or `thread.ID`. This way, you don't have a bunch of `return "", err` lines in the method, since the value is already initialized. Otherwise, I like the normal style, which can be more explicit, and therefore more readable.

Reply:
That said, I agree about the short names... seems to be a Go custom that doesn't have a good explanation.